cmd/kube-proxy/app/server.go

```diff
@@ -788,6 +787,17 @@ func (s *ProxyServer) Run() error {
 	return <-errCh
 }
 
+func getEndpointsHandlerIfRequired(proxier proxy.Provider) config.EndpointsHandler {
+	endpointsHandler, implemented := proxier.(config.EndpointsHandler)
+	if !implemented {
+		return nil
+	}
+	if endpointsHandler.SupportsEndpointSlices() {
+		klog.Fatalf("Proxier %T should not implement EndpointsHandler if it supports EndpointSlices", proxier)
```

Comment:
I like this. But i think should may be panic or exit the entire process? This is a developer error. So it should break in CI/unit testing and so on.

Reply:
`klog.Fatalf` does exit the entire process